src/main/java/com/aws/greengrass/logmanager/CloudWatchAttemptLogsProcessor.java

```diff
@@ -128,9 +128,12 @@ public CloudWatchAttempt processLogFiles(ComponentLogFileInformation componentLo
         while (!componentLogFileInformation.getLogFileInformationList().isEmpty() && !reachedMaxSize.get()) {
             LogFile logFile = componentLogFileInformation.getLogFileInformationList().get(0).getLogFile();
```

Comment:
we should stop calling `get(0)`, just get the value once, and store it in a temporary variable.